src/Package/MSBuild.UnGAC.Setup/Program.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace Microsoft.Build.UnGAC
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                string[] assembliesToUnGAC =
+                {
+                    "Microsoft.Build, Version=15.1.0.0",
+                    "Microsoft.Build.Engine, Version=15.1.0.0",
+                    "Microsoft.Build.Framework, Version=15.1.0.0",
+                    "Microsoft.Build.Tasks.Core, Version=15.1.0.0",
+                    "Microsoft.Build.Utilities.Core, Version=15.1.0.0",
+                    "Microsoft.Build.Conversion.Core, Version=15.1.0.0"
+                };
+
+                uint hresult = NativeMethods.CreateAssemblyCache(out IAssemblyCache assemblyCache, 0);
+
+                // Most significant bit is set, meaning there was an error in the Hresult.
+                if ((hresult >> 31) == 1)
+                {
+                    Console.WriteLine($"Could not successfully call CreateAssemblyCache. HResult: {hresult}");
+                    Console.WriteLine("Exiting without removing assemblies from the GAC...");
+                    return;
+                }
+
+                for (int i = 0; i < assembliesToUnGAC.Length; i++)
```

Comment:
why not a `foreach`?

Reply:
Force of habit for the _slight_ performance difference.

Reply:
C# is smart enough to emit a for loop even if you write a foreach:

https://sharplab.io/#v2:EYLgdgpgLgZgHgHwAICYCMBYAUASD37JAZgAJUy0B2bAb2x1RXrqxLfbaTQAYBtAXRIBDAM4iIAW2AAbAJYQRAFQD2AVTABxAIIBhEgF5sHDi2NnzbAEQBZWQGMATspHKYUAHQAhAK6zpAEwAaEgA1CAcRWWUwfTQAVnc0d25ky0CjC0yrW0dnVw8fP393AFEwAHNZSGCwiKiY+MTk1PTWLMybeycXNy9fAPcAMQchCQgAd2UHAGsa8Mjo2ISklO40jPbzTtyegv7ipAA2EXcdKYg5usXGlZaNzeNt7vy+ovdVKD9ZT4VT88uFg1ls01q0Hlscs9eoUBmcwAA3eb1P4OC6hJHXYGrSz3DgAXwA3PQcMSYOchHYABYkAAU8KEDmEYkkMgAniQqkzxFI5AoVOptDoAJTElj4LgAThpom5bKFRNwODx9GVWDxQA

Reply:
😲 I'm convinced!